src/rpc/misc.cpp

```diff
@@ -251,14 +252,17 @@ static RPCHelpMan deriveaddresses()
                 }
             },
             RPCExamples{
-                "First three native segwit receive addresses\n" +
+                "First three native segwit receive addresses:\n" +
                 HelpExampleCli("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\" \"[0,2]\"") +
-                HelpExampleRpc("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\", \"[0,2]\"")
+                HelpExampleRpc("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\", \"[0,2]\"") +
+                "Derive all addresses from a privkey cPsQTSmMZ8e3AEUWGjS73f5R364yJxH6RxcgnwbHjbKbFPUP2Dtu:\n" +
+                HelpExampleCli("deriveaddresses", "\"combo(cPsQTSmMZ8e3AEUWGjS73f5R364yJxH6RxcgnwbHjbKbFPUP2Dtu)\" null false")
             },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
-    RPCTypeCheck(request.params, {UniValue::VSTR, UniValueType()}); // Range argument is checked later
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValueType(), UniValue::VBOOL}); // Range argument is checked later
     const std::string desc_str = request.params[0].get_str();
+    const bool require_checksum = request.params[2].isNull() || request.params[2].get_bool();
```

Comment:
```suggestion
    const bool require_checksum{request.params[2].isNull() ? true : request.params[2].get_bool()};
```

nit: I prefer to print the default value as literal. This makes it easier to change the value without changing the logic.

Reply:
Makes sense, incorporated.